sway-core/src/semantic_analysis.rs

```diff
@@ -1,12 +1,11 @@
 //! Type checking for Sway.
 pub mod ast_node;
-pub mod namespace;
+pub mod namespace_system;
```

Comment:
I love the new `item`, `module` and `root` submodules!

That said, I'm wondering if this extra `namespace_system` might be a bit unnecessarily verbose?

What are your thoughts on keeping this as `namespace` and moving the content of what is now `namespace_system::namespace` into this module?

So e.g. rather than having

- namespace_system/mod.rs - Declares modules including namespace.
- namespace_system/namespace - Declares `Namespace`.

We could have:

- namespace/mod.rs - Declares both `Namespace` and its submodules (e.g `mod item;`, `mod module;` etc).

This might also help to keep a lot of the diff noise down by removing all the `namespace::<Type>` -> `namespace_system::<Type>` changes?

Reply:
We could do that! However, I chose to do `namespace_system/namespace.rs` and `namespace_system/mod.rs` because this:
1) eliminates the need to have a `mod.rs` file open in order to edit the `Namespace` type. This is something that @otrho mentioned that they didn't like because it was hard to tell which file was which (https://github.com/FuelLabs/sway/issues/846)
2) and also eliminates the need to have a `namespace/` subdir at the same level in the directory as the `namespace.rs` file, which is something I don't like because my text editor organizes by type and it makes it confusing (https://github.com/FuelLabs/sway/issues/846)
3) Rust doesn't like when you have files like `namespace/namespace.rs` and `namespace/mod.rs`

Reply:
Alternatively, we could name it like, `ns` or something like that

Reply:
Totally agree about it being annoying having the subdirectory at the same level as a file with the same name - my editor does the same where it first lists directories and then file names, so you need to jump around your file tree quite a bit more when hopping between files within the same module.

That said I can see where @otrho's coming from w.r.t. only seeing `mod.rs` being annoying if your editor only shows you the file name and not any of the parent path.

> Rust doesn't like when you have files like namespace/namespace.rs, so this isn't an option

Hmm I think this should be totally fine, no? That actually looks like a nice compromise that satisfies everyone's editor issues too!